### PR TITLE
Add JMH benchmarks for Executor#execute with and without context propagation.

### PR DESCRIPTION
**Pull request by ubschmidt2**

### CODE REVIEW

**Comment by adriancole** [comment]
Cool. Didnt know how to jmh agents before. Mind pasting a run from your laptop? Jmh is a tease without results ;)

**Comment by ubschmidt2** [comment]
Here's an example run (HP Z440): https://gist.github.com/ubschmidt2/f99027b7340da946b183df712334f1a1

```
Benchmark                                                       Mode     Cnt       Score   Error  Units
ExecutorInstrumentationBenchmark.automatic                    sample  339075      66.408 ± 1.967  ns/op
ExecutorInstrumentationBenchmark.automatic:automatic·p0.00    sample              40.000          ns/op
ExecutorInstrumentationBenchmark.automatic:automatic·p0.50    sample              60.000          ns/op
ExecutorInstrumentationBenchmark.automatic:automatic·p0.90    sample              77.000          ns/op
ExecutorInstrumentationBenchmark.automatic:automatic·p0.95    sample              86.000          ns/op
ExecutorInstrumentationBenchmark.automatic:automatic·p0.99    sample             115.000          ns/op
ExecutorInstrumentationBenchmark.automatic:automatic·p0.999   sample             329.848          ns/op
ExecutorInstrumentationBenchmark.automatic:automatic·p0.9999  sample           13782.176          ns/op
ExecutorInstrumentationBenchmark.automatic:automatic·p1.00    sample          116608.000          ns/op
ExecutorInstrumentationBenchmark.manual                       sample  387306      60.250 ± 0.563  ns/op
ExecutorInstrumentationBenchmark.manual:manual·p0.00          sample              34.000          ns/op
ExecutorInstrumentationBenchmark.manual:manual·p0.50          sample              63.000          ns/op
ExecutorInstrumentationBenchmark.manual:manual·p0.90          sample              70.000          ns/op
ExecutorInstrumentationBenchmark.manual:manual·p0.95          sample              72.000          ns/op
ExecutorInstrumentationBenchmark.manual:manual·p0.99          sample              87.000          ns/op
ExecutorInstrumentationBenchmark.manual:manual·p0.999         sample             267.693          ns/op
ExecutorInstrumentationBenchmark.manual:manual·p0.9999        sample            4430.463          ns/op
ExecutorInstrumentationBenchmark.manual:manual·p1.00          sample           16480.000          ns/op
ExecutorInstrumentationBenchmark.none                         sample  299609      44.420 ± 0.272  ns/op
ExecutorInstrumentationBenchmark.none:none·p0.00              sample              20.000          ns/op
ExecutorInstrumentationBenchmark.none:none·p0.50              sample              47.000          ns/op
ExecutorInstrumentationBenchmark.none:none·p0.90              sample              56.000          ns/op
ExecutorInstrumentationBenchmark.none:none·p0.95              sample              56.000          ns/op
ExecutorInstrumentationBenchmark.none:none·p0.99              sample              59.000          ns/op
ExecutorInstrumentationBenchmark.none:none·p0.999             sample              71.000          ns/op
ExecutorInstrumentationBenchmark.none:none·p0.9999            sample             270.000          ns/op
ExecutorInstrumentationBenchmark.none:none·p1.00              sample           14384.000          ns/op
```

**Comment by adriancole** [comment]
ty

File: contrib/agent/src/jmh/java/io/opencensus/contrib/agent/ExecutorInstrumentationBenchmark.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Fork;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.infra.Blackhole;
+
+/** Benchmarks for automatic context propagation added by {@link ExecutorInstrumentation}. */
+public class ExecutorInstrumentationBenchmark {
+
+  private static class MyRunnable implements Runnable {
```
Comment (bogdandrutu) [inline contrib/agent/src/jmh/java/io/opencensus/contrib/agent/ExecutorInstrumentationBenchmark.java]: final?

Comment (ubschmidt2) [inline contrib/agent/src/jmh/java/io/opencensus/contrib/agent/ExecutorInstrumentationBenchmark.java]: Thanks! Done.

**Pull request merged**